### ISSUE TITLE
[Bug] 蒸馏时的 best ckpt 无法被正确存储

### ISSUE THREAD

**Issue by Weiyun1025** [open-mmlab/mmrazor#396]
蒸馏时，teacher 和 student 会分别进行 eval，然后在 student 测试完后，checkpoint_hook 的 key_indicator 会被初始化成 student.accuracy/top1，但是在 teacher 测试完之后，checkpoint_hook 会再次触发，此时的 metrics 中没有 student.accuracy/top1，只有 teacher.accuracy/top1，此时程序会崩溃

**Comment by HIT-cwh** [comment]
Hi! I'm sorry for the inconvenience. You are right. There is no key named `student.accuracy/top1` in `teacher_metrics` and the program will fail. To solve this problem, we plan to call `after_val_epoch` hooks only once during `SingleTeacherDistillValLoop`. And here is the corresponding codes:

**Comment by Weiyun1025** [comment]
> Hi! I'm sorry for the inconvenience. You are right. There is no key named `student.accuracy/top1` in `teacher_metrics` and the program will fail. To solve this problem, we plan to call `after_val_epoch` hooks only once during `SingleTeacherDistillValLoop`. And here is the corresponding codes in [SingleTeacherDistillValLoop](https://github.com/open-mmlab/mmrazor/blob/dev-1.x/mmrazor/engine/runner/distill_val_loop.py#L46-L79):
> 
> ```
>     def run(self):
>         """Launch validation."""
>         self.runner.call_hook('before_val')
>         self.runner.call_hook('before_val_epoch')
>         self.runner.model.eval()
>         for idx, data_batch in enumerate(self.dataloader):
>             self.run_iter(idx, data_batch)
>         # compute student metrics
>         metrics = self.evaluator.evaluate(len(self.dataloader.dataset))
> 
>         self.runner.call_hook('before_val_epoch')
>         for idx, data_batch in enumerate(self.dataloader):
>             self.run_iter_teacher(idx, data_batch)
>         # compute teacher metrics
>         teacher_metrics = self.evaluator.evaluate(len(self.dataloader.dataset))
>         for key, value in teacher_metrics.items():
>             teacher_key = 'teacher.' + key
>             metrics[teacher_key] = value
> 
>         self.runner.call_hook('after_val_epoch', metrics=metrics)
>         self.runner.call_hook('after_val')
> ```
> 
> Could you help review the above codes and create a pull request for us?

Thank you for your response. I will create a pull request lately.
